pkg/kubectl/resource/interfaces.go

```diff
@@ -44,3 +44,41 @@ type ClientMapperFunc func(mapping *meta.RESTMapping) (RESTClient, error)
 func (f ClientMapperFunc) ClientForMapping(mapping *meta.RESTMapping) (RESTClient, error) {
 	return f(mapping)
 }
+
+// RequestOptions is a function that is given a chance to modify the outgoing request.
+type RequestOptions func(*client.Request)
```

Comment:
Will we ever need to return an error from this and bail?

Reply:
In the future if we do it's easy enough to change.

Reply:
ACK